R/parse.R

```diff
@@ -135,3 +138,47 @@ parse_check_consistent_dimensions_lhs <- function(eq, dat, call) {
     }
   }
 }
+
+find_ranks_in_expr <- function(expr, reduction = FALSE) {
+  ranks <- list()
+  if (is.symbol(expr)) {
+    return(ranks)
+  }
+  is_array <- any(expr[[1]] == "[")
```

Comment:
```suggestion
  is_array <- rlang::is_call(expr, "[")
```

or if you want to do it by hand, `identical(expr[[1]], quote(`[`))`